src/app/onboarding/availability/page.tsx

```diff
@@ -1,24 +1,92 @@
 'use client';
 
-import Button from '@/components/Button';
-import COLORS from '@/styles/colors';
+import { useContext, useState, useEffect } from 'react';
 import { H1 } from '@/styles/text';
 import { OnboardingContext } from '@/utils/OnboardingProvider';
-import { useContext } from 'react';
+import TextInput from '@/components/TextInput';
+import { isValidDate } from '@/utils/helpers';
+import DateInput from '@/components/DateInput';
+import { Profile } from '@/types/schema';
+import TextAreaInput from '@/components/TextAreaInput';
 
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const [hours, setHours] = useState('');
+  const [startDate, setStartDate] = useState('');
+  const [periods, setPeriods] = useState('');
+
+  useEffect(() => {
+    setHours(
+      onboarding?.profile.hours_per_month
+        ? onboarding?.profile.hours_per_month.toString()
+        : '',
+    );
+    setStartDate(onboarding?.profile.start_date || '');
+    setPeriods(onboarding?.profile.availability_description || '');
+  }, []);
```

Comment:
since we are using values of onboarding, we should include `onboarding` in the dependency array.